### PR TITLE
Add math.sumprod() , math.frexp() , and math.isclose()

### PR DESCRIPTION
**Pull request by farah-salama**
## Added functions

* `math.sumprod(x, y)`: Calculates the sum of products of values from the lists x and y.
* `math.frexp(x)`: Returns the mantissa and exponent of x as the pair (m, e), where m is a float and e is an integer such that x == m * 2**e exactly.
* `math.isclose(a, b, *, rel_tol = 1e-09, abs_tol = 0.0)`: Checks if the values a and b are close to each other.

## Testing
Tests are added in `test_math.py` to ensure the correctness of each function.

### CODE REVIEW

File: src/runtime/math.py [hunk 1]
```diff
@@ -709,3 +709,64 @@ def remainder(x: f64, y: f64) -> f64:
...
     return x - y*f64(q)
+
+
+def sumprod(x: list[f64], y: list[f64]) -> f64:
+    '''
+    Return the sum of products of values from the lists x and y.
+    '''
+    if len(x) != len(y):
+        raise ValueError("The two lists must have the same length.")
+
```
Comment (kmr-srbh) [inline src/runtime/math.py:720]: Please add the following line:
```python
i: i32
```

**Review by kmr-srbh — CHANGES_REQUESTED** [review]
Please make the requested changes for the tests to pass.

**Comment by kmr-srbh** [comment]
Hello @farah-salama! Welcome to LPython! Thanks for this PR.

Please update the test references by running 
```bash
./run_tests.py -u
```

File: src/runtime/math.py (hunk 1 above)
Comment (farah-salama) [reply src/runtime/math.py:720]: I noticed it right after I made the pr thanks for your help!

**Pull request closed without merging**